### PR TITLE
job: add isolated Job wrapper

### PR DESCRIPTION
**Pull request by tychoish**
This may be a reasonable alternative to the "blocking" execution options, but may resolve some of the concurrent job running semantic issues. Basically a single instance of a job can be made so that only one instance is running at a time (as if it were locked.)

### CODE REVIEW

**Review by reugn — COMMENTED** [review]
The idea is good. I would not require go 1.19 in the meantime. Also, worth unifying the names for the struct and the constructor. Trying to think if there is a better name for that instead of `IsolatedInstanceJob`.

File: quartz/job.go [hunk 1]
```diff
@@ -151,3 +152,28 @@ func (cu *CurlJob) Execute(ctx context.Context) {
 	cu.StatusCode = resp.StatusCode
 	cu.Response = string(body)
 }
+
+type isolatedJob struct {
+	Job
+	// TODO: switch this to an atomic.Bool when upgrading to/past go1.19
+	isRunning *atomic.Value
+}
+
```
Comment (reugn) [inline quartz/job.go:161]:
```suggestion

// Execute is called by a Scheduler when the Trigger associated with this job fires.
```

File: quartz/job_test.go [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+					// sleep for a jittered amount of
+					// time, less than 11ms
+					job.Execute(ctx)
+				case <-ctx.Done():
+					return
+				case <-sig:
+					return
+				}
+				timer.Reset(1 + time.Duration(rand.Int63n(10))*time.Millisecond)
+
```
Comment (reugn) [inline quartz/job_test.go:60]:
```suggestion
```

File: quartz/job.go [hunk 3]
```diff
@@ -151,3 +152,28 @@ func (cu *CurlJob) Execute(ctx context.Context) {
...
+	defer j.isRunning.Store(false)
+
+	j.Job.Execute(ctx)
+}
+
+// NewIsolatedJob wraps a job object and ensures that only one
+// instance of the job's Execute method can be called at a time.
+func NewIsolatedJob(j Job) Job {
+	return &isolatedJob{
+		Job:       j,
```
Comment (reugn) [inline quartz/job.go:175]:
```suggestion
func NewIsolatedJob(underlying Job) Job {
	return &isolatedJob{
		Job:       underlying,
```

File: .github/workflows/golangci-lint.yml [hunk 4]
```diff
@@ -11,6 +11,11 @@ jobs:
     name: lint
     runs-on: ubuntu-latest
     steps:
+      - name: Setup go
+        uses: actions/setup-go@v3
+        with:
+          go-version: 1.19
+
```
Comment (reugn) [inline .github/workflows/golangci-lint.yml:18]: Can be removed

Comment (tychoish) [inline .github/workflows/golangci-lint.yml:18]: ``
```

Comment (tychoish) [inline .github/workflows/golangci-lint.yml:18]:
```suggestion
```

**Pull request merged**